File: backend/app/schemas/audit.py
"""CAZZ SHIELD — Audit Schemas"""
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class AuditEventResponse(BaseModel):
    id: str
    event_id: str
    agent_id: str
    agent_name: Optional[str] = None
    department: Optional[str] = None
    action: str
    action_category: Optional[str] = None
    resource: Optional[str] = None
    decision: str
    category: str
    policy_matched: Optional[str] = None
    policy_rule: Optional[str] = None
    trust_score: Optional[float] = None
    risk_score: Optional[float] = None
    budget_status: Optional[str] = None
    budget_remaining: Optional[float] = None
    decision_path: Optional[str] = None
    operator: str
    operator_type: str
    severity: str
    record_hash: str
    prev_hash: str
    sequence_number: int
    timestamp: datetime

    class Config:
        from_attributes = True


class AuditListResponse(BaseModel):
    events: list[AuditEventResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class AuditFilters(BaseModel):
    agent_id: Optional[str] = None
    decision: Optional[str] = None
    category: Optional[str] = None
    severity: Optional[str] = None
    department: Optional[str] = None
    search: Optional[str] = None
    date_from: Optional[datetime] = None
    date_to: Optional[datetime] = None


class AuditStatsResponse(BaseModel):
    total_events: int
    total_allowed: int
    total_denied: int
    total_escalated: int
    denial_rate: float
    events_today: int
    events_this_week: int
    top_agents: list[dict]
    top_actions: list[dict]
    decision_distribution: dict
    hourly_distribution: list[dict]

File: frontend/src/lib/audit.ts
import { api } from './api';

export interface AuditEvent {
  id: string;
  event_id: string;
  agent_id: string;
  agent_name: string;
  department: string;
  action: string;
  action_category: string;
  resource: string;
  decision: string;
  category: string;
  policy_matched: string | null;
  policy_rule: string | null;
  trust_score: number;
  risk_score: number;
  budget_status: string;
  budget_remaining: number;
  decision_path: string;
  operator: string;
  operator_type: string;
  severity: string;
  record_hash: string;
  prev_hash: string;
  sequence_number: number;
  timestamp: string | null;
}

export interface AuditEventListResponse {
  events: AuditEvent[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface AuditStats {
  total_events: number;
  total_allowed: number;
  total_denied: number;
  total_escalated: number;
  denial_rate: number;
  events_today: number;
  decision_distribution: {
    allowed: number;
    denied: number;
    escalated: number;
  };
}

export const auditApi = {
  listEvents: async (params?: {
    page?: number;
    page_size?: number;
    agent_id?: string;
    decision?: string;
    category?: string;
    severity?: string;
    department?: string;
    search?: string;
    date_from?: string;
    date_to?: string;
  }) => {
    const response = await api.get<AuditEventListResponse>('/audit/events', { params });
    return response.data;
  },

  getEvent: async (eventId: string) => {
    const response = await api.get<AuditEvent>(`/audit/events/${eventId}`);
    return response.data;
  },

  getStats: async () => {
    const response = await api.get<AuditStats>('/audit/stats');
    return response.data;
  },
};
